src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -1,99 +1,84 @@
-        private volatile ITaskHandler _taskHandler;
+        private volatile ITaskHandler? _taskHandler;
 
         [ImportingConstructor]
         public TaskCenterSolutionAnalysisProgressReporter(
             SVsTaskStatusCenterService taskStatusCenterService,
             IDiagnosticService diagnosticService,
             VisualStudioWorkspace workspace)
         {
-            _lastTimeReported = DateTimeOffset.UtcNow;
-            _resettableDelay = null;
-
-            ResetProgressStatus();
-
             _taskCenterService = (IVsTaskStatusCenterService)taskStatusCenterService;
-
             _options = new TaskHandlerOptions()
             {
                 Title = ServicesVSResources.Running_low_priority_background_processes,
                 ActionsAfterCompletion = CompletionActions.None
             };
 
-            var crawlerService = workspace.Services.GetService<ISolutionCrawlerService>();
+            var crawlerService = workspace.Services.GetRequiredService<ISolutionCrawlerService>();
             var reporter = crawlerService.GetProgressReporter(workspace);
 
-            StartedOrStopped(reporter.InProgress);
+            ResetProgressStatus();
 
+            _solutionCrawlerEventHandler = new SolutionCrawlerEventHandlerWithDelay(reporter, s_minimumInterval);
             // no event unsubscription since it will remain alive until VS shutdown
-            reporter.ProgressChanged += OnSolutionCrawlerProgressChanged;
+            _solutionCrawlerEventHandler.UpdateProgress += OnUpdateProgress;
+
+            if (reporter.InProgress)
+            {
+                Started();
+            }
+            else
+            {
+                Stopped();
+            }
         }
 
-        private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progressData)
+        private void OnUpdateProgress(object sender, ProgressData progressData)
         {
-            // there is no concurrent call to this method since ISolutionCrawlerProgressReporter will serialize all
+            // there is no concurrent call to this method since SolutionCrawlerEventHandlerWithDelay will serialize all
             // events to preserve event ordering
             switch (progressData.Status)
             {
                 case ProgressStatus.Started:
-                    StartedOrStopped(started: true);
+                    Started();
                     break;
                 case ProgressStatus.PendingItemCountUpdated:
-                    _lastPendingItemCount = progressData.PendingItemCount.Value;
+                    _lastPendingItemCount = progressData.PendingItemCount ?? 0;
                     ProgressUpdated();
                     break;
                 case ProgressStatus.Stopped:
-                    StartedOrStopped(started: false);
+                    Stopped();
                     break;
                 case ProgressStatus.Evaluating:
                 case ProgressStatus.Paused:
```

Comment:
this entire switch seems totally unnecessary.  why don't we just store the ProgressData  as the latest data we get, then upate the UI (see the explanation here: https://github.com/dotnet/roslyn/pull/39457/files#r343391675)